components/hubspot/actions/common-create.mjs

```diff
@@ -0,0 +1,98 @@
+import hubspot from "../hubspot.app.mjs";
+import {
+  ObjectType, HUBSPOT_OWNER,
+} from "../common/constants.mjs";
+
+/**
+ * Returns an options method for a CRM object type, intended to be used in
+ * additionalProps
+ *
+ * @param {string} objectTypeName The object type name of the CRM object
+ * @returns The options method
+ */
+function getOptionsMethod(objectTypeName) {
+  switch (objectTypeName) {
+  case HUBSPOT_OWNER:
+    return (opts) => this.hubspot.getOwnersOptions(opts);
+  case ObjectType.COMPANY:
+    return (opts) => this.hubspot.getCompaniesOptions(opts);
+  case ObjectType.CONTACT:
+    return (opts) => this.hubspot.getContactsOptions(opts);
+  case ObjectType.DEAL:
+    return (opts) => this.hubspot.getDealsOptions(opts);
+  case ObjectType.LINE_ITEM:
+    return (opts) => this.hubspot.getLineItemsOptions(opts);
+  case ObjectType.TICKET:
+    return (opts) => this.hubspot.getTicketsOptions(opts);
+  case ObjectType.QUOTE:
+    return (opts) => this.hubspot.getQuotesOptions(opts);
+  case ObjectType.CALL:
+    return (opts) => this.hubspot.getCallsOptions(opts);
+  case ObjectType.EMAIL:
+    return (opts) => this.hubspot.getEmailsOptions(opts);
+  case ObjectType.MEETING:
+    return (opts) => this.hubspot.getMeetingsOptions(opts);
+  case ObjectType.NOTE:
+    return (opts) => this.hubspot.getNotesOptions(opts);
+  case ObjectType.TASK:
+    return (opts) => this.hubspot.getTasksOptions(opts);
+  default:
+    return () => [];
+  }
+}
+
+export default {
+  props: {
+    hubspot,
+  },
+  methods: {
+    getObjectType() {
+      throw new Error("getObjectType is not implemented");
+    },
+    isRelevantProperty(property) {
+      return !property.modificationMetadata?.readOnlyValue
+        && !property.hidden
+        && !property.label.includes("(legacy)")
+        && (!property.options || property.options.length <= 500);
+    },
+    makePropDefinition(property) {
+      let type = "string";
+      if (property.fieldType === "checkbox") {
+        type = "string[]";
+      }
+      let options = property.options?.length
+        ? property.options?.filter((o) => !o.hidden)
+          .map(({
+            label, value,
+          }) => ({
+            label,
+            value,
+          }))
+        : undefined;
```

Comment:
When a property doesn't have any options, the Hubspot API sets `property.options` to `[]` rather than `undefined`. I was setting the prop's `options` to `undefined` when `property.options` is an empty array so the prop field would be a text input field rather than an empty dropdown.